tests/qos/qos_sai_base.py

```diff
@@ -1414,20 +1415,24 @@ def updateDockerService(host, docker="", action="", service=""):  # noqa: F811
             )
             logger.info("{}ed {}".format(action, service))
 
-        """ Stop mux container for dual ToR """
-        if 'dualtor' in tbinfo['topo']['name']:
+        is_dualtor = 'dualtor' in tbinfo['topo']['name']
+        is_dualtor_active_standby = is_dualtor and active_standby_ports
+        """ Stop mux container for dual ToR Active-Standby """
+        if is_dualtor:
+            if is_dualtor_active_standby:
```

Comment:
if the dualtor is active-active, the practice is to set the active-active tors pair into active-standby so all upstream packets will only be forwarded to the active side; otherwise, the upstream packets will be ECMPed.
Please refer to https://github.com/sonic-net/sonic-mgmt/pull/16549.

Reply:
This does achieve putting the upper tor into standby state, though it'd be preferrable to use something much more simple as in 16549 and eliminate a lot of the current AS code. 
The fixtures in 16549 default to function scope, and QOS SAI needs a class scope. So the change is a bit non-trivial, requiring some more thorough testing.